features/environment.py

```diff
@@ -330,24 +355,38 @@ def after_all(context):
                 subprocess.run(["lxc", "image", "delete", image])
 
 
-def _capture_container_as_image(container_name: str, image_name: str) -> None:
-    """Capture a lxd container as an image.
+def _capture_container_as_image(
+    container_name: str,
+    image_name: str,
+    cloud_api: "Optional[pycloudlib.cloud.BaseCloud]" = None,
+) -> str:
+    """Capture a container as an image.
 
     :param container_name:
         The name of the container to be captured.  Note that this container
         will be stopped.
     :param image_name:
         The name under which the image should be published.
+    :param cloud_api: Optional pycloud BaseCloud api for applicable
+        machine_types.
     """
-    subprocess.run(["lxc", "stop", container_name])
-    subprocess.run(["lxc", "publish", container_name, "--alias", image_name])
+    if cloud_api:
+        inst = cloud_api.get_instance(container_name)
+        return cloud_api.snapshot(instance=inst)
+    else:
+        # TODO(drop this with migration to pycloudlib.lxc)
+        subprocess.run(["lxc", "stop", container_name])
+        subprocess.run(
+            ["lxc", "publish", container_name, "--alias", image_name]
+        )
+        return image_name
 
 
 def create_uat_ec2_image(context: Context, series: str) -> None:
```

Comment:
Maybe I am missing something here, but it seems that this function is no longer being used

Reply:
Right oops, dropped now.